File: app/src/main/java/com/laurelkeys/userinformationtestingrobots/userinformation/presenter/UserInformationFragmentPresenter.java
package com.laurelkeys.userinformationtestingrobots.userinformation.presenter;

import android.support.annotation.NonNull;
import android.util.Log;

import com.laurelkeys.userinformationtestingrobots.userinformation.UserInformationMVP;
import com.laurelkeys.userinformationtestingrobots.userinformation.model.UserInformation;

/**
 * Created by Tiago in July 2016.
 */
public class UserInformationFragmentPresenter implements UserInformationMVP.Presenter {

    private static final String TAG = UserInformationFragmentPresenter.class.getSimpleName();

    private UserInformationMVP.View mView;

    public UserInformationFragmentPresenter(@NonNull final UserInformationMVP.View view) {
        mView = view;
    }

    @Override
    public void onSendClick(@NonNull UserInformation userInformation) {
        Log.d(TAG, "userInformation: " + userInformation.toString());
        if (isValid(userInformation)) {
            mView.displaySuccess();
        } else {
            mView.displayFailure();
        }
    }

    private boolean isValid(@NonNull UserInformation userInformation) {
        boolean validInformation = userInformation.isValidName(userInformation.firstName)
                && userInformation.isValidName(userInformation.lastName)
                && userInformation.isValidEmail(userInformation.email)
                && userInformation.isValidAge(userInformation.age);

        return userInformation.phoneNumber == null
                ? validInformation
                : validInformation && userInformation.isValidPhoneNumber(userInformation.phoneNumber);
    }
}

File: app/src/main/java/com/laurelkeys/userinformationtestingrobots/userinformation/presenter/UserInformationFragmentPresenter.kt
package com.laurelkeys.userinformationtestingrobots.userinformation.presenter

import android.util.Log

import com.laurelkeys.userinformationtestingrobots.userinformation.UserInformationMVP
import com.laurelkeys.userinformationtestingrobots.userinformation.model.UserInformation

/**
 * Created by Tiago in July 2016.
 */
class UserInformationFragmentPresenter(private val mView: UserInformationMVP.View) : UserInformationMVP.Presenter {

    companion object { private val TAG = UserInformationFragmentPresenter::class.simpleName ?: "UserInformationFragmentPresenter" }

    override fun onSendClick(userInformation: UserInformation) {
        Log.d(TAG, "userInformation: ${userInformation.toString()}")
        if (isValid(userInformation)) {
            mView.displaySuccess()
        } else {
            mView.displayFailure()
        }
    }

    private fun isValid(userInformation: UserInformation): Boolean {
        val validInformation = userInformation.isValidName(userInformation.firstName)
                && userInformation.isValidName(userInformation.lastName)
                && userInformation.isValidEmail(userInformation.email)
                && userInformation.isValidAge(userInformation.age)

        return when (userInformation.phoneNumber) {
            null -> validInformation
            else -> validInformation && userInformation.isValidPhoneNumber(userInformation.phoneNumber)
        }
    }
}
